### PR TITLE
Prevents Scotland XLBs

### PR DESCRIPTION
**Pull request by jbarnsley10**

### CODE REVIEW

File: app/lib/validation-helpers.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
+const Joi = require('joi')
 const { isFuture, isWithinInterval, sub, addMonths, startOfDay, differenceInYears } = require('date-fns')
 const { parseDate } = require('./date-helpers')
+const { getPersonAndDogs } = require('../api/ddi-index-api/person')
 const validNewMicrochip = /^\d+$/
 
+const invalidBreedForCountryMessage = 'The address for an XL Bully dog must be in England or Wales'
```
Comment (whitewaterdesign) [inline app/lib/validation-helpers.js:7]: would this be better in the consts directory?

File: app/lib/validation-helpers.js [hunk 2]
```diff
@@ -183,10 +187,38 @@ const validateDob = (date, year) => {
...
+  }
+
+  return value
+}
+
+const validateBreedForCountryChoosingAddress = async (personReference, payload, fieldName = 'address') => {
+  const ownerAndDogs = await getPersonAndDogs(personReference)
+
+  return (ownerAndDogs.dogs.some(dog => dog.breed === 'XL Bully') && payload.address.country === 'Scotland')
+    ? new Joi.ValidationError(invalidBreedForCountryMessage, [{ message: invalidBreedForCountryMessage, path: [fieldName], type: 'custom' }])
```
Comment (whitewaterdesign) [inline app/lib/validation-helpers.js:205]: does this give you the same Joi errors that we were trying to get with the code in the endpoints?  It's way nicer. 👍

File: app/routes/cdo/edit/dog-details.js [hunk 3]
```diff
@@ -31,33 +32,34 @@ module.exports = [
       auth: { scope: anyLoggedInUser },
       handler: async (request, h) => {
         const indexNumber = request.params.indexNumber
-        const dog = await getDogDetails(indexNumber)
+        const cdo = await getCdo(indexNumber)
```
Comment (whitewaterdesign) [inline app/routes/cdo/edit/dog-details.js:36]: might be easier here to do a const dog = cdo?.dog

File: app/routes/index.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
 const { anyLoggedInUser } = require('../auth/permissions')
-const { clearCdo } = require('../session/cdo')
 
 module.exports = {
   method: 'GET',
   path: '/',
   options: {
     auth: { scope: anyLoggedInUser },
     handler: (request, h) => {
-      clearCdo(request)
```
Comment (whitewaterdesign) [inline app/routes/index.js:10]: we'll have to make sure sessions are well tested for this

**Review by whitewaterdesign — APPROVED** [review]
we should sort the ? clause, but as long as this fix goes in at some point in next day I think it's ok to merge.

File: app/routes/index.js (hunk 4 above)
Comment (jbarnsley10) [reply app/routes/index.js:10]: Yes, I've changed the tests to cover it but would be good to get Hari to test in that area

File: app/lib/validation-helpers.js (hunk 2 above)
Comment (jbarnsley10) [reply app/lib/validation-helpers.js:205]: Yes :)

File: app/lib/validation-helpers.js (hunk 1 above)
Comment (jbarnsley10) [reply app/lib/validation-helpers.js:7]: As it's a validation message, I think it better sits where it is (in a validation helper) - unless we start pulling all error messages out into a consts file

**Pull request merged**